src/System.CommandLine/ArgumentConversionResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine
+{
+    internal abstract class ArgumentConversionResult
+    {
+        private protected ArgumentConversionResult(IArgument argument)
+        {
+            Argument = argument ?? throw new ArgumentNullException(nameof(argument));
+        }
+
+        public IArgument Argument { get; }
+
+        internal string ErrorMessage { get; set; }
+
+        internal static FailedArgumentConversionResult Failure(IArgument argument, string error) => new FailedArgumentConversionResult(argument, error);
```

Comment:
Why two internal and one public?

Reply:
Looks like a historical artifact. The whole class is internal now in any case.